### DISCUSSION TITLE
dir change

### DISCUSSION THREAD

**Discussion by maxim-sedilec** [jonas-merkle/AS5047P#40] [Q&A] [answered]
how can i change sensor direction?

**Comment by jonas-merkle** [comment]
Hey @maxim-sedilec
I'm a bit confused what you mean by changing the sensor direction. Could you please describe your problem in more detail? Maybe I can help you then. 😉

Best regards
Jonas

**Comment by maxim-sedilec** [comment]
Hi, 

 

I want to build encoder with direction change, I try AS5600 and this have a hardware pin for DIR , but only 12bits resolution 

 

I’m not a(good) programmer, I don’t know if this syntaxes is correct 

 

settings1.data.values.UVW_ABI = 1; 

settings1.data.values.DIR = 1;

//

as5047p.write_SETTINGS1(settings1);

 

exit status 1

cannot convert 'AS5047P_Types::SETTINGS1_t' to 'const AS5047P_Types::SETTINGS1_t*'

thanks!

**Comment by jonas-merkle** [comment, marked as answer]
Hi @maxim-sedilec

Sorry for answering your question so late. 
To fix your compiler error you just have to replace the line `as5047p.write_SETTINGS1(settings1);` of your code with the following one: `as5047p.write_SETTINGS1(&settings1);`
It's mostly similar to your code just the "&" was added.

Best regards
Jonas
